modules/vscode-server/default.nix

```diff
@@ -1,10 +1,55 @@
-import ./module.nix ({
-  name,
-  description,
-  serviceConfig,
-}: {
-  systemd.user.services.${name} = {
-    inherit description serviceConfig;
-    wantedBy = [ "default.target" ];
-  };
-})
+import ./module.nix (
+  { name
+  , description
+  , serviceConfig
+  , lib
+  , config
+  , cfg
+  }: lib.mkMerge [
+    {
+      systemd.user.services.${name} = {
+        inherit description serviceConfig;
+        wantedBy = [ "default.target" ];
+      };
+    }
+    (lib.mkIf cfg.enableForUsers.enable {
+      systemd.tmpfiles.settings =
+        let
+          forEachUser = ({ path, file }: builtins.listToAttrs
+            (builtins.map
+              (username: let user = config.users.users.${username}; in {
```

Comment:
I'm still unable to get this to work as expected both `config.user.user.foo.name` and `"bar"` fail.
when e.g.. `config.user.user.goo.name = "bar"`
for reference `users.groups.<name>.members` handles this correctly, may be useful to take a look at that.
thank you

Reply:
Can you please expand on what doesn't "work as expected"?

Reply:
> Can you please expand on how it "doesn't work as expected"?

I have `config.users.users.main.name = "fsnkty"`
setting `services.vscode-server.enableForUsers.users = [ config.users.users.main.name ];`
breaks this in that
```
       error: attribute 'fsnkty' missing
       at /nix/store/fbgf31l3gy506di92bbw12aw95jg4cs6-source/modules/vscode-server/default.nix:20:37:
           19|             (builtins.map
           20|               (username: let user = config.users.users.${username}; in {
             |                                     ^
           21|                 name = "${user.home}/${path}";
```
giving "fsnkty" breaks in the same way, the string "main" can be used but that's non ideal and doesn't behave similarly to any other option that Ive encountered which expects a list of users.

this workaround in using the user attribute name as a string instead of its given name is non ideal because now changing that value is separate from changing it anywhere else a user is specified or named